server/services/RuleService.ts

```diff
@@ -0,0 +1,247 @@
+  GetRulesParams,
+  GetRulesResponse,
+  UpdateRuleParams,
+  UpdateRuleResponse,
+} from '../models/interfaces';
+import { CLIENT_RULE_METHODS } from '../utils/constants';
+import { Rule } from '../../models/interfaces';
+import { ServerResponse } from '../models/types';
+import { safeDump } from 'js-yaml';
+import moment from 'moment';
+
+export default class RulesService {
+  osDriver: ILegacyCustomClusterClient;
+
+  constructor(osDriver: ILegacyCustomClusterClient) {
+    this.osDriver = osDriver;
+  }
+
+  /**
+   * Calls backend POST Rules API.
+   */
+
+  createRule = async (
+    _context: RequestHandlerContext,
+    request: OpenSearchDashboardsRequest,
+    response: OpenSearchDashboardsResponseFactory
+  ): Promise<IOpenSearchDashboardsResponse<ServerResponse<CreateRuleResponse> | ResponseError>> => {
+    try {
+      const {
+        id,
+        title,
+        description,
+        detection,
+        status,
+        author,
+        references,
+        tags,
+        log_source,
+        level,
+        false_positives,
+      } = request.body as Rule;
+      const today = moment(moment.now()).format('YYYY/MM/DD');
+      const jsonPayload = {
+        id,
+        title,
+        description,
+        status,
+        author,
+        date: today,
+        modified: today,
+        references: references.map((ref) => ref.value),
+        tags: tags.map((tag) => tag.value),
+        logsource: {
+          product: log_source,
+          service: log_source,
+        },
+        level,
+        falsepositives: false_positives.map((falsePos) => falsePos.value),
+        detection: JSON.parse(detection),
+      };
+      console.log(jsonPayload);
+      const ruleYamlPayload = safeDump(jsonPayload);
+
+      const params: CreateRuleParams = { body: ruleYamlPayload, category: log_source };
+      const { callAsCurrentUser: callWithRequest } = this.osDriver.asScoped(request);
+      const createRuleResponse: CreateRuleResponse = await callWithRequest(
+        CLIENT_RULE_METHODS.CREATE_RULE,
+        params
+      );
+
+      return response.custom({
+        statusCode: 200,
+        body: {
+          ok: true,
+          response: createRuleResponse,
+        },
+      });
+    } catch (error: any) {
+      console.error('Security Analytics - RulesService - createRule:', error);
+      return response.custom({
+        statusCode: 200,
+        body: {
+          ok: false,
+          error: error.message,
+        },
+      });
+    }
+  };
+
+  getRules = async (
+    _context: RequestHandlerContext,
+    request: OpenSearchDashboardsRequest<{}, GetRulesParams>,
+    response: OpenSearchDashboardsResponseFactory
+  ): Promise<IOpenSearchDashboardsResponse<ServerResponse<GetRulesResponse> | ResponseError>> => {
+    try {
+      const { prePackaged } = request.query;
+      const params: GetRulesParams = {
+        prePackaged,
+        body: request.body,
+      };
+      console.log(`Making get rules req: ${params}`);
```

Comment:
Not blocking for this draft, but are these development logs that can be removed, or are we planning to leave this in?